plugins/emoji-resources/src/store.ts

```diff
@@ -0,0 +1,107 @@
+}
+export const getSkinTone = (): number => {
+  const skinToneKey = getEmojisLocalStorageKey('skinTone')
+  const skinTone = window.localStorage.getItem(skinToneKey)
+  if (skinTone == null) return 0
+
+  try {
+    return JSON.parse(skinTone)
+  } catch (e) {
+    console.error(e)
+    return 0
+  }
+}
+
+export const removeFrequentlyEmojis = (emoji: EmojiWithGroup): void => {
+  const hexcode = isCustomEmoji(emoji) ? emoji.shortcode : emoji.hexcode
+  if (hexcode === undefined) return
+
+  const frequentlyEmojisKey = getEmojisLocalStorageKey()
+  const frequentlyEmojis = window.localStorage.getItem(frequentlyEmojisKey)
+  if (frequentlyEmojis != null) {
+    const parsedEmojis = JSON.parse(frequentlyEmojis)
+    if (Array.isArray(parsedEmojis)) {
+      window.localStorage.setItem(
+        frequentlyEmojisKey,
+        JSON.stringify(parsedEmojis.filter((pe) => pe.hexcode !== hexcode))
+      )
+    }
+  }
+}
+export const addFrequentlyEmojis = (emoji: EmojiWithGroup): void => {
+  if (emoji === undefined) return
+  const hexcode = isCustomEmoji(emoji) ? emoji.shortcode : emoji.hexcode
+
+  const frequentlyEmojisKey = getEmojisLocalStorageKey()
+  const frequentlyEmojis = window.localStorage.getItem(frequentlyEmojisKey)
+  const empty = frequentlyEmojis == null
+
+  if (!empty) {
+    const parsedEmojis = JSON.parse(frequentlyEmojis)
+    if (Array.isArray(parsedEmojis)) {
+      const index = parsedEmojis.findIndex((pe) => pe.hexcode === hexcode)
+      if (index === -1) parsedEmojis.push({ hexcode, count: 1 })
+      else parsedEmojis[index].count++
+      parsedEmojis.sort((a, b) => b.count - a.count)
+      window.localStorage.setItem(frequentlyEmojisKey, JSON.stringify(parsedEmojis))
+      return undefined
+    }
+  }
+  window.localStorage.setItem(frequentlyEmojisKey, JSON.stringify([{ hexcode, count: 1 }]))
+}
+export const getFrequentlyEmojis = (): EmojiWithGroup[] | undefined => {
+  const frequentlyEmojisKey = getEmojisLocalStorageKey()
+  const frequentlyEmojis = window.localStorage.getItem(frequentlyEmojisKey)
+  if (frequentlyEmojis == null) return undefined
+
+  try {
+    const parsedEmojis = JSON.parse(frequentlyEmojis)
+    if (!Array.isArray(parsedEmojis)) return undefined
+    const emojis = get(emojiStore)
+    const result: EmojiWithGroup[] = []
+    emojis.forEach((emoji: EmojiWithGroup) => {
+      if (isCustomEmoji(emoji)) {
+        if (parsedEmojis.find((pe) => pe.hexcode === emoji.shortcode) !== undefined) result.push(emoji)
+      } else {
+        parsedEmojis.forEach((parsedEmoji: any) => {
+          if (parsedEmoji.hexcode === emoji.hexcode) {
+            result.push(emoji)
+            return
+          }
+          const skinEmoji = emoji.skins?.find((s) => s.hexcode === parsedEmoji.hexcode)
+          if (skinEmoji === undefined) return
+          result.push({ ...skinEmoji, key: '' })
+        })
+      }
+    })
+    return result
+  } catch (e) {
+    console.error(e)
```

Comment:
[Minor] Suggest to add error message:
`console.error('Failed to parse emojis', e)`